### 작업형 1

mtcars 데이터셋(mtcars.csv)의 qsec 컬럼을 최소최대 척도(Min-Max Scale)로 변환한 후 0.5보다
큰 값을 가지는 레코드 수를 구하시오

In [2]:
import pandas as pd

In [3]:
mtcars = pd.read_csv('mtcars.csv')

# 작업유형 2

In [28]:
x_train = pd.read_csv('dataset/x_train.csv',encoding = 'euc_kr')
x_test = pd.read_csv('dataset/X_test.csv',encoding = 'euc_kr')
y_train = pd.read_csv('dataset/y_train.csv',encoding = 'euc_kr')
print(x_train.shape ,y_train.shape , x_test.shape)

(3500, 10) (3500, 2) (2482, 10)


### EDA

In [29]:
desc_t = x_train.describe(include='all').T
desc_t

count unique   top  freq         mean          std          min  \
cust_id  3500    NaN   NaN   NaN       1749.5      1010.51            0   
총구매액     3500    NaN   NaN   NaN  9.19193e+07  1.63507e+08 -5.24215e+07   
최대구매액    3500    NaN   NaN   NaN  1.96642e+07  3.19923e+07   -2.992e+06   
환불금액     1205    NaN   NaN   NaN  2.40782e+07  4.74645e+07         5600   
주구매상품    3500     42    기타   595          NaN          NaN          NaN   
주구매지점    3500     24  본  점  1077          NaN          NaN          NaN   
내점일수     3500    NaN   NaN   NaN      19.2537      27.1749            1   
내점당구매건수  3500    NaN   NaN   NaN      2.83496      1.91237            1   
주말방문비율   3500    NaN   NaN   NaN     0.307246     0.289752            0   
구매주기     3500    NaN   NaN   NaN      20.9583      24.7487            0   

                 25%          50%          75%          max  
cust_id       874.75       1749.5      2624.25         3499  
총구매액     4.74705e+06  2.82227e+07  1.06508e+08  2.32318e+09  
최대구매액      2.875e+06    9.837e+06  2.29625e+07  7.06629e+08  
환불금액       2.259e+06    7.392e+06    2.412e+07  5.63753e+08  
주구매상품            NaN          NaN          NaN          NaN  
주구매지점            NaN          NaN          NaN          NaN  
내점일수               2            8           25          285  
내점당구매건수      1.66667      2.33333        3.375      22.0833  
주말방문비율     0.0272909      0.25641      0.44898            1  
구매주기               4           13           28          166

In [30]:
x_train.isnull().sum()

cust_id       0
총구매액          0
최대구매액         0
환불금액       2295
주구매상품         0
주구매지점         0
내점일수          0
내점당구매건수       0
주말방문비율        0
구매주기          0
dtype: int64

- 환불금액에 2295개의 null값이 존재. -> 최소환불금액이 5600원이므로, 환불안한사람은 null 을 가지고있음. fillna(0) 처리
- 주구매상품, 주구매지점 -> 카테고리 컬럼이므로, one-hot encoding 을 적용


In [31]:
x_train=x_train.fillna(0)
x_test=x_test.fillna(0)

In [32]:
x_train=pd.get_dummies(x_train, columns = ['주구매상품', '주구매지점'], drop_first = True)
x_test=pd.get_dummies(x_test, columns = ['주구매상품', '주구매지점'], drop_first = True)

In [41]:
print(x_train.shape ,y_train.shape , x_test.shape)
set(list(x_train.columns)) - set(list(x_test.columns))

(3500, 71) (3500, 2) (2482, 71)


set()

In [40]:
# x_test는 없으므로, 학습을 시키지 않음 
x_train = x_train.drop('주구매상품_소형가전', axis = 1)

In [43]:
y_train

cust_id  gender
0           0       0
1           1       0
2           2       1
3           3       1
4           4       0
...       ...     ...
3495     3495       1
3496     3496       1
3497     3497       0
3498     3498       0
3499     3499       0

[3500 rows x 2 columns]